roles/adoption_osp_deploy/tasks/prepare_undercloud.yml

```diff
@@ -0,0 +1,242 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure controller and undercloud vms are started and reachable
+  vars:
+    _cifmw_libvirt_manager_layout:
+      vms:
+        controllers:
+          start: true
+          disk_file_name: "dummy"
+        osp-underclouds:
+          start: true
+          disk_file_name: "dummy"
+    cifmw_libvirt_manager_all_vms: >-
+      {%- set vms = {} -%}
+      {%- for vm in _vm_groups['osp-underclouds'] -%}
+        {%- set _ = vms.update({vm: 'osp-underclouds'}) -%}
+      {%- endfor -%}
+      {%- for vm in _vm_groups['controllers'] -%}
+        {%- set _ = vms.update({vm: 'controllers'}) -%}
+      {%- endfor -%}
+      {{ vms }}
+  ansible.builtin.include_role:
```

Comment:
ansible_libvirt_pools fact expected by in start_vms.yml is not defined at this point and must be defined. 
I see there is task "Gather pool fact" in libvirt_manage role which builds this variable. Maybe it there should be the same task before starting tasks from start_vms.yml

Reply:
added the var, thanks